File: deployments/01_aci/deploy_to_aci.py
from azureml.core import Environment, Model, Workspace
from azureml.core.model import InferenceConfig
from azureml.core.webservice import AciWebservice

ws = Workspace.from_config(path="../../")
model = Model(
  workspace=ws,
  name="diabetes-sklearn",
  version=1,
)

env = Environment.from_conda_specification(
  name="diabetes-sklearn-aci",
  file_path="./conda.yml",
)

# env.register(workspace=ws)
# env.build(workspace=ws)

config = InferenceConfig(
  source_directory="./src",
  entry_script="score.py",
  environment=env,
)

aci_config = AciWebservice.deploy_configuration(
  cpu_cores=1,
  memory_gb=2,
)

endpoint = Model.deploy(
  workspace=ws,
  name="diabetes-aci-deploy",
  models=[model],
  inference_config=config,
  deployment_config=aci_config,
  overwrite=True,
)

endpoint.wait_for_deployment(show_output=True)
print(endpoint.state)

File: r/deployments/01_aci/deploy_to_aci.R
library(azuremlsdk)

ws <- load_workspace_from_config(path = "../../../")
model <- get_model(
  workspace = ws,
  name = "accidents-logistic",
  version = as.integer(1)
)

env <- r_environment(
  name = "r-environment",
  custom_docker_image = "9054a1a822d04d21a7a5f4e726da20cc.azurecr.io/r-azureml:latest" # nolint
)

inference_config <- inference_config(
  source_directory = "./src",
  entry_script = "score.R",
  environment = env
)

deployment_config <- aci_webservice_deployment_config(
  cpu_cores = 1,
  memory_gb = 1
)

deployment_name <- "accidents-aci"
current_deployment <- tryCatch({
    get_webservice(
    workspace = ws,
    name = deployment_name
  )
}, error = function(e) {
  current_deployment <- ""
})

if (current_deployment == "") {
  endpoint <- deploy_model(
    workspace = ws,
    name = deployment_name,
    models = list(model),
    inference_config = inference_config,
    deployment_config = deployment_config
  )

  wait_for_deployment(
    webservice = endpoint,
    show_output = TRUE
  )
} else {
  update_aci_webservice(
    webservice = current_deployment,
    models = list(model),
    inference_config = inference_config
  )

  wait_for_deployment(
    webservice = current_deployment,
    show_output = TRUE
  )
}
